central/group/datastore/datastore_impl.go

```diff
@@ -378,3 +393,24 @@ func (ds *dataStoreImpl) getDefaultGroupForProps(ctx context.Context, props *sto
 	// 2. Filter for the default group.
 	return ds.getByProps(ctx, &storage.GroupProperties{AuthProviderId: props.GetAuthProviderId()})
 }
+
+// validateMutableGroupIDNoLock validates whether a group allows changes or not based on the mutability mode set.
+// NOTE: This function assumes that the call to this function is already behind a lock.
+func (ds *dataStoreImpl) validateMutableGroupIDNoLock(ctx context.Context, id string, force bool) error {
+	group, exists, err := ds.storage.Get(ctx, id)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return errox.NotFound.Newf("group with id %q was not found", id)
+	}
+
+	if group.GetProps().GetTraits().GetMutabilityMode() == storage.Traits_ALLOW_MUTATE {
```

Comment:
Please see [my comment on the other PR](https://github.com/stackrox/stackrox/pull/2610#discussion_r946295009) on proper enforcement of mutability modes